imports/ui/components/study_plan/TabbedContainer.jsx

```diff
@@ -0,0 +1,327 @@
+          setPlannerName(plannerID, userInput);
+        }
+      }
+    }
+  }
+
+  /**
+   * Handler for when user clicks outside of the input field when editing a
+   * study plan and entering a new name for this
+   *
+   * @param {[String]} plannerID    Planner ID of study plan to be edited
+   * @event {[Object]} event    Event object of this HTML object
+   */
+  handleCancelEditStudyPlan(plannerID, event) {
+    this.setState({ isEditingPlanName: false });
+
+    // If the user already entered a name, add study plan for convenience
+    if (isDefinedObj(event)) {
+      let userInput = event.target.value;
+      if (userInput) {
+        setPlannerName(plannerID, userInput);
+      }
+    }
+  }
+
+  //===================================================================
+  // RENDER FUNCTIONS FOR DISTINCT UI PARTS WITHIN THE TABBEDCONTAINER
+  //===================================================================
+  /**
+   * Renders a tab with its internal UI components and event handlers.
+   * Tabs contain a caret that activates a dropdown menu and event handlers
+   * for selecting an active tab, editing or deleting a study plan.
+   *
+   * @param {[String]} tabTitle    Text of tab's title
+   * @param {[int]} index    Index of this tab. From the left onscreen, the
+   *                         indices start from 0.
+   * @param {[String]} plannerID     ID of this tab's study plan
+   * @return    React component representing a Tab.
+   */
+  renderTab(tabTitle, index, plannerID) {
+    let enabledMouseOver = true;
+    let tabTitleComponent = tabTitle;
+
+    if (this.state.planNameToEdit == plannerID && this.state.isEditingPlanName) {
+      // Enable the input field only for the study plan the user wants to edit
+      enabledMouseOver = false;
+      tabTitleComponent = this.renderTabTitleInput(tabTitle, plannerID);
+    } else {
+      tabTitleComponent = this.renderTabTitle(tabTitle, plannerID)
+    }
+
+    return (
+      <Tab key={index} tabWidth="9em"
+           navSpanClass="nav-link-in" navSpanStyle={{position: "relative"}}
+           tabTitle={tabTitleComponent}
+           enabledMouseOver={true}
+           isFirstTab={(index === 0) ? true : false}
+           onClickTab={this.handleClickTab.bind(this, index)}
+           onClickDeleteTab={this.handleDeleteStudyPlanClick.bind(this, index)}
+           onClickEditTab={this.handleEditStudyPlan.bind(this, plannerID)}
+           isActiveTab={(this.state.tabSelectedIndex === index)} />
+    )
+  }
+
+  /**
+   * Renders the title of a tab. If the number of characters for this tab
+   * is too long (>MAX_STUDY_PLAN_NAME_LENGTH chars), it will be sliced so
+   * that tab titles don't contain Strings that are too long to fit in a tab.
+   *
+   * @param {[String]} tabTitle    Text of tab's title
+   * @param {[String]} plannerID     ID of this tab's study plan
+   * @return    Shortened String representation of this tab title.
+   */
```

Comment:
Yup! Usually @param and @return get highlighted on most text editors (even on GitHub)! So it looks clearer too!